File: binary_search/go/binary_search.go
package main

import "fmt"

func main() {
	d := []int{1, 2, 3, 4, 5}

	fmt.Println(bin_search(d, 3))
}
func bin_search(d []int, number int) int {
	low := 0
	max := len(d) - 1
	for low <= max {
		mid := (low + max) / 2
		guess := d[mid]
		if guess == number {
			return mid
		} else if guess > number {
			max = mid - 1
		} else if guess < number {
			low = mid + 1
		}
	}
	return 0
}

File: binary_search/binary_search.py
def bin_search(arr, item):
    low = 0
    maxi = len(arr) - 1

    while low <= maxi:
        mid1: int  = (low + maxi) // 2
        guess = arr[mid1]
        if guess == item:
            return mid1
        elif guess > item:
             maxi = mid1 - 1
        else:
            low = mid1 + 1
    return None
my_list = [1,2,3,4,5,6,7,8]
print(my_list)
print(bin_search(my_list,6))

